NETOBSERV-1981: Fix Exporters form view

Amoghrd opened the pull request:
## Description
Fix form view for exporters section

## Dependencies
n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [x] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labeled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [x] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

jpinsonneau reviewed (approved):
Code looks good, did you give a try ?

Amoghrd commented:
nope not yet. Will try it now

Amoghrd commented:
Do the changes to files other than the CSV look good?

jpinsonneau commented on config/crd/bases/flows.netobserv.io_flowcollectors.yaml:
```diff
@@ -1155,10 +1155,11 @@ spec:
                         type: object
                       type:
-                        description: '`type` selects the type of exporters. The available options are `KAFKA` and `IPFIX`.'
+                        description: '`type` selects the type of exporters. The available options are `KAFKA`, `IPFIX`, and `OpenTelemetry`.'
```
That one is not needed as OpenTelemetry has been introduced in v1beta2

You can see it here: https://github.com/netobserv/network-observability-operator/blob/main/config/crd/bases/flows.netobserv.io_flowcollectors.yaml#L5641

Amoghrd replied:
yeah thats what I thought. Thats why wanted to double check. Thanks for review

jpinsonneau commented on docs/FlowCollector.md:
```diff
@@ -2277,9 +2277,9 @@ only the result of this request.<br/>
         <td>
-          `type` selects the type of exporters. The available options are `KAFKA` and `IPFIX`.<br/>
+          `type` selects the type of exporters. The available options are `KAFKA`, `IPFIX`, and `OpenTelemetry`.<br/>
           <br/>
-            <i>Enum</i>: KAFKA, IPFIX<br/>
+            <i>Enum</i>: KAFKA, IPFIX, OpenTelemetry<br/>
```
same here this is aboute v1beta1

Amoghrd commented:
/ok-to-test

Amoghrd commented:
OpenTelemtry does not show up as default now but I dont see the sub fields for any of the exporters so there is still something missing
<img width="871" alt="Screenshot 2024-11-26 at 11 18 13 PM" src="https://github.com/user-attachments/assets/d2da94a6-fbcf-4221-b47d-6b3f4097b615">

Amoghrd commented:
/ok-to-test

Amoghrd commented on config/csv/bases/netobserv-operator.clusterserviceversion.yaml:
```diff
@@ -277,10 +277,26 @@ spec:
           path: exporters[0].ipfix
           x-descriptors:
             - "urn:alm:descriptor:com.tectonic.ui:fieldDependency:exporters.type:IPFIX"
+        - displayName: IPFIX TargetHost
```
okay this change did not help. Let me know what I am missing here

jpinsonneau replied:
I'm not sure why but these doesn't work anymore on arrays :disappointed: 
https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md#20-fielddependency

Why not getting rid of these and simply keep:
```
          - description: additional optional exporters for custom consumption or storage.
            path: exporters
            x-descriptors:
              - 'urn:alm:descriptor:com.tectonic.ui:advanced'
          - displayName: Type
            path: 'exporters[0].type'
```

jpinsonneau replied:
![image](https://github.com/user-attachments/assets/2026e14e-863e-49f4-a993-e87025d3fe95)

Amoghrd replied:
Lets see if this helps

Amoghrd replied:
yeah so pushed your suggestion so looks as above.
But the flowcollector is not getting created due to error
<img width="886" alt="Screenshot 2024-11-27 at 7 15 10 PM" src="https://github.com/user-attachments/assets/e174efe4-5458-4a5e-a045-00e0e29cc84e">

Since the YAML view looks as below
<img width="278" alt="Screenshot 2024-11-27 at 7 15 30 PM" src="https://github.com/user-attachments/assets/cceecc1c-da7b-4585-a5e7-7e27bbac14c7">

jpinsonneau replied:
![abe-simpson-abe-simpson-cloud](https://github.com/user-attachments/assets/02cd476a-dd11-4abf-b12c-0af1f05739a7)

It seems like there is an issue with default values and this legacy auto generated form.
We may add default values to kafka address + topic and openTelemetry targetHost + targetPort to solve that.

@jotak thoughts on that ?

Amoghrd commented:
Another issue I see is when I select KAFKA exporter in form view and click on YAML view I see OpenTelemetry still present
![Uploading Screenshot 2024-11-26 at 11.39.36 PM.png…]()

Amoghrd commented:
/ok-to-test

jotak commented:
@Amoghrd after changing the base CSV you need to run `make update-bundle`

Amoghrd commented:
I am getting error when I run make update-bundle

Error: not all generators ran successfully
run `controller-gen rbac:roleName=manager-role crd:crdVersions=v1 paths=./... output:crd:artifacts:config=config/crd/bases output:webhook:dir=./config/webhook webhook -w` to see all available markers, or `controller-gen rbac:roleName=manager-role crd:crdVersions=v1 paths=./... output:crd:artifacts:config=config/crd/bases output:webhook:dir=./config/webhook webhook -h` for usage
make: *** [manifests] Error 1

But these files get updated

modified:   apis/flowcollector/v1beta1/zz_generated.deepcopy.go
	modified:   apis/flowcollector/v1beta2/zz_generated.deepcopy.go
	modified:   apis/flowmetrics/v1alpha1/zz_generated.deepcopy.go
	modified:   config/crd/bases/flows.netobserv.io_flowcollectors.yaml
	modified:   config/crd/bases/flows.netobserv.io_flowmetrics.yaml
	modified:   config/rbac/role.yaml

Amoghrd commented:
/ok-to-test

Amoghrd commented:
@jotak Deleted the `bin/controller-gen` and `bin/operator-sdk` and could see the `bundle/manifests/netobserv-operator.clusterserviceversion.yaml` change still. But did not get any errors when I ran `make update-bundle`.
But the above error screenshot wrt the other exporters section still seen even when not selected is not rectified. So dont know how to go about this

jotak commented on bundle/manifests/netobserv-operator.clusterserviceversion.yaml:
```diff
@@ -469,7 +469,7 @@ spec:
       specDescriptors:
       - description: defines the desired type of deployment for flow processing.
-        displayName: Deployment model
+        displayName: udeployment LModel
```
not sure how you did that, but all those names look pretty funky !

Amoghrd replied:
yeah I just reverted the changes to that file

Amoghrd commented:
/ok-to-test

Amoghrd commented:
Tried the first fix with `make update-bundle` still same issue as above

Amoghrd commented:
/ok-to-test

Amoghrd commented:
@jotak The exporters view is still not fixed with latest changes and still facing the issue mentioned [here](https://github.com/netobserv/network-observability-operator/pull/912#discussion_r1860694471)

jotak commented:
@Amoghrd I rebased and fixed the bad generated names issue, can you check if this is still good for you?

jpinsonneau commented:
are we good to merge this ?

Amoghrd commented:
I have attached the screen-recording of the issue I am still facing on the JIRA. I am not able to figure out how to fix that. If that will be fixed at a later stage then this PR can be merged to fix the view

Amoghrd commented:
/ok-to-test

Amoghrd commented:
Okay, that helped and I dont see the error now. Thanks @jpinsonneau 
This can be merged if code looks good

Amoghrd commented:
Only Konflux check is failing. Will merge the PR
/approve
